gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +969,920 @@ public void testListDnsRecords() {
+      try {
+        result.get();
+        fail("Zone does not exist yet");
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(404, ex.code());
+        assertFalse(ex.retryable());
+      }
+      // zone exists but has no changes
+      DNS.create(ZONE1);
+      batch = DNS.batch();
+      result = batch.listChangeRequests(ZONE1.name());
+      batch.submit();
+      ImmutableList<ChangeRequest> changes = ImmutableList.copyOf(result.get().iterateAll());
+      assertEquals(1, changes.size()); // default change creating SOA and NS
+      // zone has changes
+      ChangeRequest change = DNS.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1);
+      waitForChangeToComplete(ZONE1.name(), change.generatedId());
+      change = DNS.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1);
+      waitForChangeToComplete(ZONE1.name(), change.generatedId());
+      batch = DNS.batch();
+      result = batch.listChangeRequests(ZONE1.name());
+      DnsBatchResult<Page<ChangeRequest>> errorPageSize = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.pageSize(0));
+      DnsBatchResult<Page<ChangeRequest>> errorPageNegative = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.pageSize(-1));
+      DnsBatchResult<Page<ChangeRequest>> resultAscending = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING));
+      DnsBatchResult<Page<ChangeRequest>> resultDescending = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.DESCENDING));
+      DnsBatchResult<Page<ChangeRequest>> resultAdditions = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.ADDITIONS));
+      DnsBatchResult<Page<ChangeRequest>> resultDeletions = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.DELETIONS));
+      DnsBatchResult<Page<ChangeRequest>> resultId = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.ID));
+      DnsBatchResult<Page<ChangeRequest>> resultTime = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.START_TIME));
+      DnsBatchResult<Page<ChangeRequest>> resultStatus = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.STATUS));
+      batch.submit();
+      changes = ImmutableList.copyOf(result.get().iterateAll());
```

Comment:
Also here you could use [Iterables.size](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Iterables.html#size(java.lang.Iterable))